core/src/main/java/org/openapitools/openapidiff/core/compare/PathsDiff.java

```diff
@@ -134,9 +136,21 @@ private static boolean parametersIntersect(List<Parameter> left, List<Parameter>
     int parametersSize = left.size();
     long intersectedParameters =
         IntStream.range(0, left.size())
-            .filter(
-                i -> left.get(i).getSchema().getType().equals(right.get(i).getSchema().getType()))
+            .filter(i -> parametersTypeEquals(left.get(i), right.get(i)))
             .count();
     return parametersSize == intersectedParameters;
   }
+
+  /**
+   * Checks is provided parameter pair equal by type and format
+   *
+   * @param left parameter from the first compared method
+   * @param right parameter from the second compared method
+   * @return <code>true</code> in case parameter pair is of the same type; <code>false</code>
+   *     otherwise
+   */
+  private static boolean parametersTypeEquals(Parameter left, Parameter right) {
+    return Objects.equals(left.getSchema().getType(), right.getSchema().getType())
+        && Objects.equals(left.getSchema().getFormat(), right.getSchema().getFormat());
```

Comment:
[question] I'm assuming `getFormat` returns OK where one isn't explicitly specified, correct?

Reply:
In case format is not specified explicitly, it will be parsed as null. Objects#equals method is null safe, so it will compare nullable values correctly.